network/p2p/network_test.go

```diff
@@ -5,309 +5,245 @@ package p2p
-			requestFunc: func(network *Network) error {
-				return network.AppRequest(context.Background(), ids.GenerateTestNodeID(), 0, time.Time{}, []byte{unregistered})
-			},
-			err: nil,
-		},
-		{
-			name: "drop empty app request message",
-			requestFunc: func(network *Network) error {
-				return network.AppRequest(context.Background(), ids.GenerateTestNodeID(), 0, time.Time{}, []byte{})
-			},
-			err: nil,
+			name: "nil",
+			msg:  nil,
 		},
 		{
-			name: "drop unregistered cross-chain app request message",
-			requestFunc: func(network *Network) error {
-				return network.CrossChainAppRequest(context.Background(), ids.GenerateTestID(), 0, time.Time{}, []byte{unregistered})
-			},
-			err: nil,
+			name: "empty",
+			msg:  []byte{},
 		},
 		{
-			name: "drop empty cross-chain app request message",
-			requestFunc: func(network *Network) error {
-				return network.CrossChainAppRequest(context.Background(), ids.GenerateTestID(), 0, time.Time{}, []byte{})
-			},
-			err: nil,
-		},
-		{
-			name: "drop unregistered gossip message",
-			requestFunc: func(network *Network) error {
-				return network.AppGossip(context.Background(), ids.GenerateTestNodeID(), []byte{unregistered})
-			},
-			err: nil,
-		},
-		{
-			name: "drop empty gossip message",
-			requestFunc: func(network *Network) error {
-				return network.AppGossip(context.Background(), ids.GenerateTestNodeID(), []byte{})
-			},
-			err: nil,
-		},
-		{
-			name: "drop unrequested app request failed",
-			requestFunc: func(network *Network) error {
-				return network.AppRequestFailed(context.Background(), ids.GenerateTestNodeID(), 0)
-			},
-			err: ErrUnrequestedResponse,
+			name: "non-empty",
+			msg:  []byte("foobar"),
 		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+			ctx := context.Background()
+			handler := &testHandler{
+				appGossipF: func(context.Context, ids.NodeID, []byte) {
+					require.Fail("should not be called")
+				},
+				appRequestF: func(context.Context, ids.NodeID, time.Time, []byte) ([]byte, error) {
+					require.Fail("should not be called")
+					return nil, nil
+				},
+				crossChainAppRequestF: func(context.Context, ids.ID, time.Time, []byte) ([]byte, error) {
+					require.Fail("should not be called")
+					return nil, nil
+				},
+			}
+			network := NewNetwork(logging.NoLog{}, nil, prometheus.NewRegistry(), "")
+			_, err := network.NewAppProtocol(handlerID, handler)
+			require.NoError(err)
+
+			require.Nil(network.AppRequest(ctx, ids.EmptyNodeID, 0, time.Time{}, []byte("foobar")))
+			require.Nil(network.AppGossip(ctx, ids.EmptyNodeID, []byte("foobar")))
+			require.Nil(network.CrossChainAppRequest(ctx, ids.Empty, 0, time.Time{}, []byte("foobar")))
+		})
+	}
+}
+
+// A response or timeout for a request we never made should return an error
+func TestResponseForUnrequestedRequest(t *testing.T) {
+	tests := []struct {
+		name string
+		msg  []byte
```

Comment:
`msg` seems unused

Reply:
Whoops... blaming this on being tired...